### PR TITLE
ipc: dai: extend Zephyr native drivers api to ipc/dai.c

### PR DESCRIPTION
**Pull request by abonislawski**
This patch will extend Zephyr native drivers api usage
to ipc dai files and full flow with dma release will be supported

### CODE REVIEW

File: src/ipc/ipc3/dai.c [hunk 1]
```diff
@@ -257,7 +257,11 @@ void dai_dma_release(struct comp_dev *dev)
 	if (dd->chan) {
 		/* remove callback */
 		notifier_unregister(dev, dd->chan, NOTIFIER_ID_DMA_COPY);
+#if CONFIG_ZEPHYR_NATIVE_DRIVERS
```
Comment (abonislawski) [inline src/ipc/ipc3/dai.c:260]: @lgirdwood  one open: not sure If we want to support ipc3 dai with native drivers?

Comment (lgirdwood) [reply src/ipc/ipc3/dai.c:260]: > @lgirdwood one open: not sure If we want to support ipc3 dai with native drivers?

Yes we need to support IPC3 as there is no current TSC decision to deprecate IPC3

Comment (lgirdwood) [reply src/ipc/ipc3/dai.c:260]: > @lgirdwood do we plan to deprecate IPC3 in the future in favor of IPC4?

That's a TSC decision, the if/when/how of IPC deprecation and planning. Good topic for next TSC though.

File: src/ipc/ipc4/dai.c [hunk 2]
```diff
@@ -229,7 +237,7 @@ static void dai_dma_position_init(struct dai_data *dd)
 
 int dai_config(struct comp_dev *dev, struct ipc_config_dai *common_config,
 	       void *spec_config)
-{trace_point(9);
+{
```
Comment (lgirdwood) [inline src/ipc/ipc4/dai.c:240]: oh, looks like this sneaked in....

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
CI is good. @juimonen good for you too ?

**Review by kv2019i — APPROVED** [review]
As a late comment to https://github.com/thesofproject/sof/pull/5610 , the intent of CONFIG_ZEPHYR_NATIVE_DRIVERS is not very clear. The description implies this is a working option, but in practise this option cannot be used to build a fully functional image until all uses of DMA support both Zephyr and legacy (i.e. this and follow PRs).

But with that confusion still present in the tree, this code seems good to go.

**Comment by abonislawski** [comment]
@kv2019i that was the goal for CONFIG_ZEPHYR_NATIVE_DRIVERS to use extra kconfig until everything works perfectly, otherwise we would just use __ZEPHYR__
For now we have DMA transfers in host and dai components, so already pretty functional :)

**Pull request merged**